wpilibc/src/main/native/cpp/ADIS16448_IMU.cpp

```diff
@@ -154,6 +154,65 @@ ADIS16448_IMU::ADIS16448_IMU(IMUAxis yaw_axis, SPI::Port port,
   m_connected = true;
 }
 
+ADIS16448_IMU::ADIS16448_IMU(ADIS16448_IMU&& other)
+    : m_reset_in{std::move(other.m_reset_in)},
+      m_status_led{std::move(other.m_status_led)},
+      m_yaw_axis{std::move(other.m_yaw_axis)},
+      m_gyro_rate_x{std::move(other.m_gyro_rate_x)},
+      m_gyro_rate_y{std::move(other.m_gyro_rate_y)},
+      m_gyro_rate_z{std::move(other.m_gyro_rate_z)},
+      m_accel_x{std::move(other.m_accel_x)},
+      m_accel_y{std::move(other.m_accel_y)},
+      m_accel_z{std::move(other.m_accel_z)},
+      m_mag_x{std::move(other.m_mag_x)},
+      m_mag_y{std::move(other.m_mag_y)},
+      m_mag_z{std::move(other.m_mag_z)},
+      m_baro{std::move(other.m_baro)},
+      m_temp{std::move(other.m_temp)},
+      m_tau{std::move(other.m_tau)},
+      m_dt{std::move(other.m_dt)},
+      m_alpha{std::move(other.m_alpha)},
+      m_compAngleX{std::move(other.m_compAngleX)},
+      m_compAngleY{std::move(other.m_compAngleY)},
+      m_accelAngleX{std::move(other.m_accelAngleX)},
+      m_accelAngleY{std::move(other.m_accelAngleY)},
+      m_offset_buffer{other.m_offset_buffer},
+      m_gyro_rate_offset_x{std::move(other.m_gyro_rate_offset_x)},
+      m_gyro_rate_offset_y{std::move(other.m_gyro_rate_offset_y)},
+      m_gyro_rate_offset_z{std::move(other.m_gyro_rate_offset_z)},
+      m_avg_size{std::move(other.m_avg_size)},
+      m_accum_count{std::move(other.m_accum_count)},
+      m_integ_gyro_angle_x{std::move(other.m_integ_gyro_angle_x)},
+      m_integ_gyro_angle_y{std::move(other.m_integ_gyro_angle_y)},
+      m_integ_gyro_angle_z{std::move(other.m_integ_gyro_angle_z)},
+      m_thread_active{other.m_thread_active.load()},
+      m_first_run{other.m_first_run.load()},
+      m_thread_idle{other.m_thread_idle.load()},
+      m_start_up_mode{other.m_start_up_mode.load()},
+      m_auto_configured{std::move(other.m_auto_configured)},
+      m_spi_port{std::move(other.m_spi_port)},
+      m_calibration_time{std::move(other.m_calibration_time)},
+      m_spi{std::move(other.m_spi)},
+      m_auto_interrupt{std::move(other.m_auto_interrupt)},
+      m_connected{std::move(other.m_connected)},
+      m_acquire_task{std::move(other.m_acquire_task)},
+      m_simDevice{std::move(other.m_simDevice)},
+      m_simConnected{std::move(other.m_simConnected)},
+      m_simGyroAngleX{std::move(other.m_simGyroAngleX)},
+      m_simGyroAngleY{std::move(other.m_simGyroAngleZ)},
+      m_simGyroAngleZ{std::move(other.m_simGyroAngleZ)},
+      m_simGyroRateX{std::move(other.m_simGyroRateX)},
+      m_simGyroRateY{std::move(other.m_simGyroRateY)},
+      m_simGyroRateZ{std::move(other.m_simGyroRateZ)},
+      m_simAccelX{std::move(other.m_simAccelX)},
+      m_simAccelY{std::move(other.m_simAccelY)},
+      m_simAccelZ{std::move(other.m_simAccelZ)},
+      m_mutex{std::move(other.m_mutex)} {}
+
+ADIS16448_IMU ADIS16448_IMU::operator=(ADIS16448_IMU&& other) {
+  return ADIS16448_IMU(std::move(other));
```

Comment:
This doesn't do the correct thing. It creates a new instance, moves into it, and returns it. It should move into the current object instead.

Copy-paste the fields above and use std::swap() on them. Also add a check to the top of the move assignment operator like this:
```cpp
if (this == &other) {
  return *this;
}
```